File: android/src/main/java/expo/modules/gaodemap/overlays/CircleViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Circle 视图 Module
 */
class CircleViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("CircleView")

    View(CircleView::class) {
      Events("onPress")
      
      Prop<Map<String, Double>>("center") { view, center ->
        view.setCenter(center)
      }
      
      Prop<Double>("radius") { view, radius ->
        view.setRadius(radius)
      }
      
      Prop<Int>("fillColor") { view, color ->
        view.setFillColor(color)
      }
      
      Prop<Int>("strokeColor") { view, color ->
        view.setStrokeColor(color)
      }
      
      Prop<Float>("strokeWidth") { view, width ->
        view.setStrokeWidth(width)
      }

      Prop<Float>("zIndex") { view, zIndex ->
        view.setZIndex(zIndex)
      }
    }
  }
}

File: ios/overlays/CircleViewModule.swift
import ExpoModulesCore

public class CircleViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("CircleView")
        
        View(CircleView.self) {
            Events("onPress")
            
            Prop("center") { (view: CircleView, center: [String: Double]) in
                view.setCenter(center)
            }
            
            Prop("radius") { (view: CircleView, radius: Double) in
                view.setRadius(radius)
            }
            
            Prop("fillColor") { (view: CircleView, color: String) in
                view.setFillColor(color)
            }
            
            Prop("strokeColor") { (view: CircleView, color: String) in
                view.setStrokeColor(color)
            }
            
            Prop("strokeWidth") { (view: CircleView, width: Double) in
                view.setStrokeWidth(Float(width))
            }
        }
    }
}